From jason.wessel@windriver.com  Tue May 12 10:00:57 2009
From: Jason Wessel <jason.wessel@windriver.com>
Date: Mon, 11 May 2009 15:24:10 -0500
Subject: USB: serial: ftd_sio: implement sysrq handling on break
To: greg@kroah.com
Cc: stern@rowland.harvard.edu, Jason Wessel <jason.wessel@windriver.com>
Message-ID: <1242073450-3794-5-git-send-email-jason.wessel@windriver.com>


Change driver to make use of the new functions in
include/linux/usb/serial.h so as to allow the driver to handle the
sysrq

Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ftdi_sio.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -2093,6 +2093,7 @@ static void ftdi_process_read(struct wor
 		if (data[packet_offset+1] & FTDI_RS_BI) {
 			error_flag = TTY_BREAK;
 			dbg("BREAK received");
+			usb_serial_handle_break(port);
 		}
 		if (data[packet_offset+1] & FTDI_RS_PE) {
 			error_flag = TTY_PARITY;
@@ -2107,8 +2108,11 @@ static void ftdi_process_read(struct wor
 				/* Note that the error flag is duplicated for
 				   every character received since we don't know
 				   which character it applied to */
-				tty_insert_flip_char(tty,
-					data[packet_offset + i], error_flag);
+				if (!usb_serial_handle_sysrq_char(port,
+						data[packet_offset + i]))
+					tty_insert_flip_char(tty,
+						data[packet_offset + i],
+						error_flag);
 			}
 			need_flip = 1;
 		}
